### ISSUE TITLE
Invalid Haskell generated

### ISSUE THREAD

**Issue by ndmitchell** [awakesecurity/proto3-suite#97]
Given:

```

syntax = "proto3";
package daml_lf_1;

message Foo {
  oneof Bar {
    sfixed64 Baz = 5;
  }
}
```

This generates:

```
((Hs.coerce @(Hs.Int64)
                                         @(HsProtobuf.Signed (HsProtobuf.Fixed Hs.Int64)))
                                        (HsProtobuf.ForceEmit y))
```

Which is ill-typed as the second argument is `ForceEmit Int64`, not `Int64`, leading to the error:

```
gen\DamlLf1.hs:52:42: error:
    * Couldn't match expected type `Hs.Int64'
                  with actual type `HsProtobuf.ForceEmit Hs.Int64'
    * In the third argument of `Hs.coerce', namely
        `(HsProtobuf.ForceEmit y)'
      In the second argument of `HsProtobuf.encodeMessageField', namely
        `((Hs.coerce
             @(Hs.Int64) @(HsProtobuf.Signed (HsProtobuf.Fixed Hs.Int64)))
            (HsProtobuf.ForceEmit y))'
      In the expression:
        (HsProtobuf.encodeMessageField
           (HsProtobuf.FieldNumber 5)
           ((Hs.coerce
               @(Hs.Int64) @(HsProtobuf.Signed (HsProtobuf.Fixed Hs.Int64)))
              (HsProtobuf.ForceEmit y)))
   |
52 |                [case fooBar of
   |                                          ^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by ndmitchell** [comment]
For a more complete example, which is what I was using as souce, see https://github.com/digital-asset/daml/blob/master/daml-lf/archive/da/daml_lf_1.proto and the 3 other files in that directory.

**Comment by themattchan** [comment]
Resolved by #99
